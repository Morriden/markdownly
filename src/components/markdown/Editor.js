import React from 'react';
import styles from './Editor.css';
import { useMarkdown, useDispatch } from '../../hooks/appContext';
import { updateMarkdown } from '../../utils/actions';

const Editor = () => {
  const markdown = useMarkdown();
  const dispatch = useDispatch();

  return (
    <textarea className={styles.Editor} value={markdown} onChange={({ target }) => dispatch(updateMarkdown(target.value))} />
  );
};

export default Editor;
