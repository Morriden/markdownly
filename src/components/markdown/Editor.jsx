import React from 'react';
import { useDispatch, useSelector } from '../../hooks/appContext';
import { getMarkdown } from '../../selectors/markdownSelectors';
import { updateMarkdown } from '../../actions/actions';
import styles from './Editor.css';
import { useParams } from 'react-router-dom';

const Editor = () => {
  const dispatch = useDispatch();
  const { id = 1234 } = useParams();
  const markdown = useSelector(getMarkdown, +id);

  return (
    <textarea 
      className={styles.Editor} 
      value={markdown} 
      onChange={({ target }) => 
        dispatch(updateMarkdown(target.value, +id))} 
    />
  );
};

export default Editor;
