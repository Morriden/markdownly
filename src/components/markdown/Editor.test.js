import React from 'react';
import { shallow } from 'enzyme';
import Editor from './Editor';
import AppProvider from '../../provider/AppProvider';

describe('Editor component', () => {
  it('renders Editor', () => {
    const wrapper = shallow(<AppProvider><Editor /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
