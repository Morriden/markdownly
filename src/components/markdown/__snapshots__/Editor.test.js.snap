// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component renders Editor 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "markdown": "Im a test sentence",
      },
    }
  }
>
  <Editor />
</ContextProvider>
`;
